openedx/api.py

```diff
@@ -97,6 +99,13 @@ def create_edx_user(user):
             user=user, platform=PLATFORM_EDX
         )
 
+        if open_edx_user.edx_username is None:
+            if edx_username is None:
+                # If we don't have an edx username provided, generate one from the user's email
+                edx_username = user.email
+            open_edx_user.edx_username = edx_username
+            open_edx_user.save()
```

Comment:
This ends up sending `None` as the username below (on line 132). Probably best to reference the `open_edx_user` directly there; that seemed to work best in testing.

I'm still getting a 403 error from Open edX when testing this locally so not sure what's going on there. I did check to make sure the payload that is going to edX looks more-or-less right but I can't get it to be happy with the request. Not sure if this is a misconfiguration issue on my end or something else.

Reply:
I added the reference to the edx user. I think that cached property doesn't get updated properly. I had same issue  in the tests.